src/include/duckdb/parser/expression/window_expression.hpp

```diff
@@ -27,6 +27,33 @@ enum class WindowBoundary : uint8_t {
 
 const char *ToString(WindowBoundary value);
 
+static ExpressionType WindowToExpressionType(string &fun_name) {
+	if (fun_name == "rank") {
+		return ExpressionType::WINDOW_RANK;
+	} else if (fun_name == "rank_dense" || fun_name == "dense_rank") {
+		return ExpressionType::WINDOW_RANK_DENSE;
+	} else if (fun_name == "percent_rank") {
+		return ExpressionType::WINDOW_PERCENT_RANK;
+	} else if (fun_name == "row_number") {
+		return ExpressionType::WINDOW_ROW_NUMBER;
+	} else if (fun_name == "first_value" || fun_name == "first") {
+		return ExpressionType::WINDOW_FIRST_VALUE;
+	} else if (fun_name == "last_value" || fun_name == "last") {
+		return ExpressionType::WINDOW_LAST_VALUE;
+	} else if (fun_name == "nth_value" || fun_name == "last") {
```

Comment:
Heh you have copied and pasted a bug here!  It makes more sense to export the original from `transform_function.cpp`

Reply:
Tried this, but including transform.hpp in an R extension caused compiler errors since transform.hpp includes files from the libpg_parser. 
I imagine we don't want the R tool importing files from the Postgres parse?

Reply:
Could also move the function definition to `window_expression.hpp` as part of WindowExpression